flocker/node/agents/blockdevice.py

```diff
@@ -1116,7 +1224,46 @@ def shared_state_changes(self):
         These are the only parts of the state that need to be sent to the
         control service.
         """
-        return (self.node_state, self.nonmanifest_datasets)
+        manifestations = {}
```

Comment:
So this function creates `devices` and `paths` from low-level data even though `discover_state()` already had that information in the `RawState` object. Couldn't `discover_state` just pass those in?